zebra-scan/src/storage.rs

```diff
@@ -0,0 +1,50 @@
+//! Store viewing keys and results of the scan.
+
+use std::collections::HashMap;
+
+use zebra_chain::{block::Height, transaction::Hash};
+
+/// Store key info and results of the scan.
+#[allow(dead_code)]
+pub struct Storage {
+    /// The key and an optional birthday for it.
+    keys: HashMap<String, Option<Height>>,
+
+    /// The key and the related transaction id.
+    results: HashMap<String, Vec<Hash>>,
```

Comment:
Is `Hash` a reasonable thing to store here? Should we also store block height where the transaction is ?

Reply:
I don't think we need to store the block height or hash. Using the transaction hash, we can get the block height and hash from the state, so there's no need to store it here as well.

**New Ticket:** Here's how we could do that state request, I will open a ticket for it.

We can use the transaction hash to look up `tx_loc_by_hash`, which gives us a `TransactionLocation` containing the block height (and transaction index). And we can get the block hash from the height. We have `read::find` methods for some of this already.
https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/state-db-upgrades.md#current-state-database-format
https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/service/read/find.rs#L129

Reply:
The state request is now ticket #7943.